src/base_images.ml

```diff
@@ -60,7 +60,14 @@ let main config mode channel capnp_address github_auth submission_uri staging_pa
     let channel = Option.map read_channel_uri channel in
     let staging_auth = staging_password_file |> Option.map (fun path -> staging_user, read_first_line path) in
     run_capnp capnp_address >>= fun (vat, rpc_engine_resolver) ->
-    let submission_cap = Capnp_rpc_unix.Vat.import_exn vat submission_uri in
+    let submission_cap =
+      match Option.map (Capnp_rpc_unix.Vat.import_exn vat) submission_uri with
+      | Some cap -> cap
+      | None ->
```

Comment:
I think it would be better to be explicit about turning this off. The service really isn't useful without a builder and it could be confusing if it just says it's trying to connect all the time.

I wonder if perhaps `Capnp_rpc_unix.Vat.import_exn` should accept `NULL` or something. Then you could use `--submission-service=NULL` or similar any time you wanted to test something without setting up a real service. Though in many cases it would be better to give an error instead of just hanging.

Reply:
Is it definitely useful enough to push it back up to `capnp_rpc_unix`? The first time you're running it, it's handy to be able to get to the "oh yes, a pipeline diagram!" without having to explain getting the cluster started up, but that's only because the git cloning and so forth can be done earlier.

Another option might be to display a warning that there's no submission service, so it's just going to pretend? Purely from a tutorial perspective, it's neat not to have to specify much on the command line, but beyond that...